patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch

```diff
@@ -0,0 +1,73 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Anton Tayanovskyy <anton@pulumi.com>
+Date: Tue, 23 Apr 2024 13:57:01 -0400
+Subject: [PATCH] Fix panic in aws_directory_service_directory Read
+
+Per API docs VpcSettings is not always set, so accessing it can cause a nil panic.
+
+One case where this seems to happen (but difficult to reproduce) is importing a shared directory, see:
+
+https://github.com/pulumi/pulumi-aws/issues/1152
+
+diff --git a/internal/service/ds/directory.go b/internal/service/ds/directory.go
+index 33252ddb7d..9d283612cb 100644
+--- a/internal/service/ds/directory.go
++++ b/internal/service/ds/directory.go
+@@ -295,6 +295,11 @@ func resourceDirectoryRead(ctx context.Context, d *schema.ResourceData, meta int
+ 	if err != nil {
+ 		return sdkdiag.AppendErrorf(diags, "reading Directory Service Directory (%s): %s", d.Id(), err)
+ 	}
++	return resourceDirectoryReadDescription(d, dir)
++}
++
++func resourceDirectoryReadDescription(d *schema.ResourceData, dir *directoryservice.DirectoryDescription) diag.Diagnostics {
++	var diags diag.Diagnostics
+ 
+ 	d.Set("access_url", dir.AccessUrl)
+ 	d.Set("alias", dir.Alias)
+@@ -315,10 +320,16 @@ func resourceDirectoryRead(ctx context.Context, d *schema.ResourceData, meta int
+ 	d.Set("edition", dir.Edition)
+ 	d.Set("enable_sso", dir.SsoEnabled)
+ 	d.Set("name", dir.Name)
++
+ 	if aws.StringValue(dir.Type) == directoryservice.DirectoryTypeAdconnector {
+ 		d.Set("security_group_id", dir.ConnectSettings.SecurityGroupId)
+-	} else {
++	} else if dir.VpcSettings != nil {
+ 		d.Set("security_group_id", dir.VpcSettings.SecurityGroupId)
++	} else {
++		// From VpcSettings docs: this member is only present if the directory is a Simple AD or Managed
++		// Microsoft AD directory; possibly the user is trying to import a shared directory, but TF still
++		// requires a Computed to be set.
++		d.Set("security_group_id", nil)
+ 	}
+ 	d.Set("short_name", dir.ShortName)
+ 	d.Set("size", dir.Size)
+diff --git a/internal/service/ds/directory_internal_test.go b/internal/service/ds/directory_internal_test.go
+new file mode 100644
+index 0000000000..3651319f90
+--- /dev/null
++++ b/internal/service/ds/directory_internal_test.go
+@@ -0,0 +1,22 @@
++// Copyright (c) HashiCorp, Inc.
++// SPDX-License-Identifier: MPL-2.0
++
++package ds
++
++import (
++	"testing"
++
++	"github.com/aws/aws-sdk-go/service/directoryservice"
++)
++
++func TestResourceDirectoryReadDoesNotPanicOnMissingVpcSettings(t *testing.T) {
```

Comment:
Do we/should we run upstream tests in our CI?

Reply:
I'll look into it. We should but possibly don't.

Reply:
I'm pretty certain we don't for GCP, so likely we don't for AWS - I imagine some of our patches won't pass tests, so likely to be some effort to make them work.

Just wanted to make sure you are not surprised by this.

Reply:
It think it's an effort worth taking for our team.

Reply:
https://github.com/pulumi/pulumi-aws/issues/3022 yes it is still out there. I'll reprioritize this